docs/configuration.md

```diff
@@ -1,9 +1,92 @@
 ## Parameter Configuration
 
-*todo: document YAML config.*
+The parameters that the software uses for analysis of the data can be configured in a [YAML](https://yaml.org/) file. An example configuration file is included in the directory `config/example.yml`. You do not need to edit the code to change the parameters.
 
-The parameters that the software uses for analysis of the data can be configured in `Config.ini` by simply opening the file in a text editor and changing the variables. You do not need to edit the code to change the parameters.
+If no configuration YAML file is found while running TopoStats, it will generate a default file.
 
-When updating TopoStats, the `Config.ini` file is ignored, so your parameters are maintained. Different sets of parameters can be saved for different sample types in the config file, and the sections for the different sample types are labelled in square brackets.
+### Example
 
-If no config file is found while running TopoStats, it will make a copy of the default config file.
\ No newline at end of file
+```{yaml}
+base_dir: ./tests/
+output_dir: ./output
+save_plots: true
+warnings: ignore
+cores: 4
+quiet: false
+file_ext: .spm
+channel: Height
+amplify_level: 1.0
+filter:
+  threshold:
+    method: std_dev # Options : otsu, std_dev, absolute
+    otsu_multiplier: 1.0
+    std_dev: 1.0
+    absolute:
+      - 1
+      - -1
+grains:
+  gaussian_size: 2.0
+  gaussian_mode: nearest
+  absolute_smallest_grain_size: 100
+  background: 0.0
+  zrange: [0, 3] # low and high height range
+  mask_direction: upper # Options: upper, lower (currently only applied to thresh+mask image)
+  threshold:
+    method: std_dev # Options : otsu, std_dev, absolute
+    otsu_multiplier: 1.0
+    std_dev: 1.0
+    absolute:
+      - 1
+      - -1
+plotting:
+  save: True
+  colorbar: True
+  cmap: nanoscope # Options : nanoscope, afmhot
+```
+
+### Core parameters
+
+| Parameter | Description | Example | Allowable |
+| --- | --- | --- | --- |
+| `base_dir` | The directory (folder) in which the software should search for images to analyse. | `./tests/` | Path. |
+| `output_dir` | The directory (folder) in which output should be saved. | `./output` | Path. |
+| `save_plots` | Should plots be saved? | `true` | `true` or `false` |
```

Comment:
This and line 90 seem very similar. This is good for now but maybe we need to think about condensing the config file and avoid confusion in the future, and check what each of these do?